soc/ite/ec/it8xxx2/Kconfig

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020 ITE Corporation. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_ITE_IT8XXX2
+	select CPU_HAS_FPU if "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "zephyr" || RISCV_ISA_EXT_M
+	select SOC_FAMILY_ITE_EC
+	select HAS_PM
+
+config SOC_IT8XXX2
+	select RISCV
+	select ATOMIC_OPERATIONS_BUILTIN
+	select RISCV_ISA_RV32I
+	select RISCV_ISA_EXT_ZICSR
+	select RISCV_ISA_EXT_ZIFENCEI
+	# Workaround mul instruction bug, see:
+	# https://www.ite.com.tw/uploads/product_download/it81202-bx-chip-errata.pdf
+	select RISCV_ISA_EXT_M if !(SOC_IT81302_BX || SOC_IT81202_BX)
+	select RISCV_ISA_EXT_A
+	select RISCV_ISA_EXT_C
+	select FLASH
+	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_DRIVER_ENABLED
```

Comment:
Were these in the original board file? Seems they were, that is odd

Reply:
Let's remove these and `HAS_FLASH_LOAD_OFFSET` since they should not be required and flash load offset is 0

Reply:
interesting, seems like this should be moved in `drivers/flash/Kconfig.it8xxx2` but maybe it's better to leave it here for now so these patches are confined under `soc/`

Reply:
Removed, not moved :D there is no reason the flash driver must be enabled for this board that I can see

Reply:
okay:

```
        select FLASH
        select FLASH_HAS_PAGE_LAYOUT
        select FLASH_HAS_DRIVER_ENABLED
        select HAS_FLASH_LOAD_OFFSET
```

all gone